sinks/signalfx/signalfx.go

```diff
@@ -208,69 +210,36 @@ func (sfx *SignalFxSink) Flush(ctx context.Context, interMetrics []samplers.Inte
 	return err
 }
 
-// FlushEventsChecks sends events to SignalFx. It does not support checks.
+var successSpanTags = map[string]string{"sink": "signalfx", "results": "success"}
+var failureSpanTags = map[string]string{"sink": "signalfx", "results": "failure"}
+
+// FlushOtherSamples sends events to SignalFx. It does not support checks.
 func (sfx *SignalFxSink) FlushOtherSamples(ctx context.Context, samples []ssf.SSFSample) {
 	span, _ := trace.StartSpanFromContext(ctx, "")
 	defer span.ClientFinish(sfx.traceClient)
-
 	for _, sample := range samples {
-
-		if _, ok := sample.Tags[dogstatsd.EventIdentifierKey]; !ok {
-			// This isn't an event, just continue
-			continue
-		}
-
-		dims := map[string]string{}
-
-		// Copy common dimensions in
-		for k, v := range sfx.commonDimensions {
-			dims[k] = v
-		}
-		// And hostname
-		dims[sfx.hostnameTag] = sfx.hostname
-
-		for k, v := range sample.Tags {
-			if k == dogstatsd.EventIdentifierKey {
-				// Don't copy this tag
-				continue
+		switch ddOtherSampleKind(sample.Tags) {
+		case ddSampleEvent:
+			ev := sfx.reportEvent(&sample)
+			// TODO: Split events out the same way as points
+			sfx.defaultClient.AddEvents(ctx, []*event.Event{&ev})
+			// report evt
+		case ddSampleServiceCheck:
+			err := sfx.convertServiceCheck(&sample)
+			spanTags := successSpanTags
```

Comment:
hm, doesn't look like these are tags that go on a span... should probably just be called `tags`?

Reply:
I think it's added to the span a few lines down.

Reply:
In changing this to a post-loop reporting process I'm removing this variable however I'm not sure that I've understood this comment.

Reply:
Hm, I meant that we don't modify the tags of the span itself - only the tags on the metric (which we add to the span, but that doesn't affect the span's tags)